samples/tutorial/Tutorial2.md

```diff
@@ -0,0 +1,483 @@
+        val welcomeScreen = context.renderChild(WelcomeWorkflow) { output ->
+          // When WelcomeWorkflow emits LoggedIn, turn it into our login action.
+          login(output.name)
+        }
+        return welcomeScreen
+      }
+
+      // When the state is Todo, defer to the TodoListWorkflow.
+      is Todo -> {
+        val todoScreen = context.renderChild(TodoListWorkflow, props = ListProps(state.name)) {
+          logout()
+        }
+        return todoScreen
+      }
+    }
+  }
+
+  // …
+}
+```
+
+### Props (TODO better title?)
+
+So far we are gathering a name from the welcome screen, and there's a place on the todo list screen to display the name, but we're not passing the name from the welcome screen to the next one. We'll fix this by passing the name down from the `RootWorkflow` to the `TodoListWorkflow` via "props".
+
+TK
+
+This works, but with no animation between the two screens it's pretty unsatisfying. We'll fix that by using a different "container" to provide the missing transition animation.
+
+### Back Stack and "Containers"
+
+We want to animate changes between our screens. Because we want all of our navigation state to be declarative, we need to use the [`BackStackScreen`](https://square.github.io/workflow/kotlin/api/workflow/com.squareup.workflow1.ui.backstack/-back-stack-screen/) to do this:
+
+```kotlin
+class BackStackScreen<StackedT : Any>(
+  bottom: StackedT,
+  rest: List<StackedT>
+) {
+  // …
+
+  val frames: List<StackedT> = listOf(bottom) + rest
+
+  // …
+}
+```
+
+The `BackStackScreen` contains a list of all screens in the back stack that are specified on each render pass. Update the `RootWorkflow` to return a `BackStackScreen` with a list of back stack items:
+
+```kotlin
+object RootWorkflow : StatefulWorkflow<Unit, State, Nothing, BackStackScreen<Any>>() {
+
+  // …
+
+  @OptIn(WorkflowUiExperimentalApi::class)
+  override fun render(
+    props: Unit,
+    state: State,
+    context: RenderContext
+  ): BackStackScreen<Any> {
+
+    // Our list of back stack items. Will always include the "WelcomeScreen".
+    val backstackScreens = mutableListOf<Any>()
+
+    // Render a child workflow of type WelcomeWorkflow. When renderChild is called, the
+    // infrastructure will create a child workflow with state if one is not already running.
+    val welcomeScreen = context.renderChild(WelcomeWorkflow) { output ->
+      // When WelcomeWorkflow emits LoggedIn, turn it into our login action.
+      login(output.name)
+    }
+    backstackScreens += welcomeScreen
+
+    when (state) {
+      // When the state is Welcome, defer to the WelcomeWorkflow.
+      is Welcome -> {
+        // We always add the welcome screen to the backstack, so this is a no op.
+      }
+
+      // When the state is Todo, defer to the TodoListWorkflow.
+      is Todo -> {
+        backstackScreens.addAll(context.renderChild(TodoListWorkflow, props = ListProps(state.name)) {
```

Comment:
Confused by `addAll` and `.frames` here, wondering if you skipped a step. It looks like this code assumes that `TodoListWorkflow` also renders `BackStackScreen`, but we haven't made such a change.

Reply:
I refactored some code, and forgot to refactor the tutorial. Fixed.